flexmeasures/data/models/planning/tests/test_solver.py

```diff
@@ -272,25 +272,49 @@ def test_fallback_to_unsolvable_problem(target_soc, charging_station_name):
     )
 
 
+@pytest.mark.parametrize(
+    "market_scenario",
+    [
+        "dynamic contract",
+        "fixed contract",
+    ],
+)
 def test_building_solver_day_2(
     db,
     add_battery_assets,
     add_market_prices,
+    create_test_tariffs,
     add_inflexible_device_forecasts,
     inflexible_devices,
     flexible_devices,
+    market_scenario: str,
 ):
-    """Check battery scheduling results within the context of a building with PV, for day 2,
-    which is set up with 8 expensive, then 8 cheap, then again 8 expensive hours.
-    We expect the scheduler to:
-    - completely discharge within the first 8 hours
-    - completely charge within the next 8 hours
-    - completely discharge within the last 8 hours
+    """Check battery scheduling results within the context of a building with PV, for day 2, against the following market scenarios:
+    1) a dynamic tariff with equal consumption and feed-in tariffs, that is set up with 8 expensive, then 8 cheap, then again 8 expensive hours.
+    2) a fixed consumption tariff and a fixed feed-in tariff that is lower.
```

Comment:
add: "which incentives to maximize self-consumption of PV power into the battery"